pkgs/applications/science/math/caffe/default.nix

```diff
@@ -0,0 +1,70 @@
+{ stdenv
+, openblas
+, boost
+, cudaSupport ? true
+, cudnnSupport ? false
+, cudnn ? null
+, cudatoolkit7
+, fetchzip
+, google-gflags
+, glog
+, hdf5
+, leveldb
+, lmdb
+, opencv
+, protobuf
+, snappy
+}:
+
+
+let optional = stdenv.lib.optional;
+in stdenv.mkDerivation rec {
+  # Use git revision because latest "release" is really old
+  name = "caffe-git-2015-07-02";
+  rev = "77d66dfc";
+
+  src = fetchzip {
```

Comment:
I've used that before but I was told to [use tarballs instead](https://github.com/NixOS/nixpkgs/pull/6907#discussion_r26823367), I'm guessing fetchFromGithub is preferred for revisions?